src/modules/peek/Peek.FilePreviewer/Previewers/UnsupportedFilePreviewer/UnsupportedFilePreviewer.cs

```diff
@@ -22,69 +22,74 @@
+
+        /// <summary>
+        /// The maximum view updates per second when enumerating a folder's contents.
+        /// </summary>
+        private const int MaxUpdateFps = 15;
+
+        /// <summary>
+        /// The icon to display when a file or folder's thumbnail or icon could not be retrieved.
+        /// </summary>
+        private static readonly SvgImageSource DefaultIcon = new(new Uri("ms-appx:///Assets/Peek/DefaultFileIcon.svg"));
 
         [ObservableProperty]
-        private UnsupportedFilePreviewData preview = new UnsupportedFilePreviewData();
+        private UnsupportedFilePreviewData preview = new();
 
         [ObservableProperty]
         private PreviewState state;
 
         public UnsupportedFilePreviewer(IFileSystemItem file)
         {
-            _folderSizeDispatcherTimer.Interval = TimeSpan.FromMilliseconds(500);
-            _folderSizeDispatcherTimer.Tick += FolderSizeDispatcherTimer_Tick;
-
             Item = file;
-            Preview.FileName = file.Name;
-            Preview.DateModified = file.DateModified?.ToString(CultureInfo.CurrentCulture);
             Dispatcher = DispatcherQueue.GetForCurrentThread();
         }
 
         private IFileSystemItem Item { get; }
 
         private DispatcherQueue Dispatcher { get; }
 
-        private Task<bool>? IconPreviewTask { get; set; }
-
-        private Task<bool>? DisplayInfoTask { get; set; }
-
-        public void Dispose()
-        {
-            _folderSizeDispatcherTimer.Tick -= FolderSizeDispatcherTimer_Tick;
-            GC.SuppressFinalize(this);
-        }
-
-        public Task<PreviewSize> GetPreviewSizeAsync(CancellationToken cancellationToken)
-        {
-            Size? size = new Size(680, 500);
-            var previewSize = new PreviewSize { MonitorSize = size, UseEffectivePixels = true };
-            return Task.FromResult(previewSize);
-        }
+        public Task<PreviewSize> GetPreviewSizeAsync(CancellationToken cancellationToken) =>
+            Task.FromResult(new PreviewSize { MonitorSize = new Size(680, 500), UseEffectivePixels = true });
 
         public async Task LoadPreviewAsync(CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
+            try
+            {
+                await Dispatcher.RunOnUiThread(async () =>
+                {
+                    Preview.FileName = Item.Name;
+                    Preview.DateModified = Item.DateModified?.ToString(CultureInfo.CurrentCulture);
 
-            State = PreviewState.Loading;
+                    State = PreviewState.Loaded;
 
-            IconPreviewTask = LoadIconPreviewAsync(cancellationToken);
-            DisplayInfoTask = LoadDisplayInfoAsync(cancellationToken);
+                    await LoadIconPreviewAsync(cancellationToken);
+                });
 
-            await Task.WhenAll(IconPreviewTask, DisplayInfoTask);
+                var progress = new Progress<string?>(update =>
```

Comment:
Can this be changed to `Progress<string>`?

Reply:
Thanks. Done.